### ISSUE TITLE
Session permission errors during initial setup

### ISSUE THREAD

**Issue by Poorchop** [electerious/Lychee#547]
### Detailed description of the problem
I can't get past the initial setup screen because of a permission error in writing to `/var/lib/php/session`. I'm running an nginx server and this directory is owned by apache. I've circumvented this problem in the past by changing `session.save_path` to `/tmp` but this doesn't seem to be working. I can't tell whether Lychee is ignoring my settings in php.ini and php-fpm.d/www.conf or if I just set something up incorrectly along the way.

### Steps to reproduce the issue
1. Follow the instructions in [Installation.md](https://github.com/electerious/Lychee/blob/master/docs/Installation.md)
2. Using nginx, use the setup described in #106 as per the installation guide
3. In php.ini, set the following: `session.save_path = "/tmp"` and ensure that the owner and group are both set to `nginx`
4. In php-fpm.d/www.conf, set the following: `php_value[session.save_path] = /tmp`
5. Ensure that SELinux write permissions are correctly set on Lychee/data and Lychee/uploads
6. Restart php-fpm and nginx
7. Follow the instructions on the setup screen, only to run into the permission error again

### Output of the diagnostics (Settings => Diagnostics)
I haven't gotten past the initial setup screen so I don't think I can view this. Here is my error log though:

```logtalk
2016/05/18 01:05:47 [error] 28901#0: *2 FastCGI sent in stderr: "PHP message: PHP Warning:  session_start(): open(/var/lib/php/session/sess_d67f2rbu9pa18v4o62hc8ju6o4, O_RDWR) failed: Permission denied (13) in /var/www/Lychee/php/index.php on line 40" while reading response header from upstream, client: 104.148.188.210, server: pictures.example.com, request: "POST /php/index.php HTTP/1.1", upstream: "fastcgi://127.0.0.1:9000", host: "pictures.example.com"
2016/05/18 01:05:47 [error] 28901#0: *2 FastCGI sent in stderr: "PHP message: PHP Warning:  Unknown: open(/var/lib/php/session/sess_d67f2rbu9pa18v4o62hc8ju6o4, O_RDWR) failed: Permission denied (13) in Unknown on line 0
PHP message: PHP Warning:  Unknown: Failed to write session data (files). Please verify that the current setting of session.save_path is correct (/var/lib/php/session) in Unknown on line 0" while reading upstream, client: 104.148.188.210, server: pictures.example.com, request: "POST /php/index.php HTTP/1.1", upstream: "fastcgi://127.0.0.1:9000", host: "pictures.example.com"
```

### Browser and system
Firefox 46.0.1 (client) and Fedora 23 (host)

After many hours, I feel like I'm inching closer to a working installation but I just can't figure this out after many additional hours of searching. Thanks for any help.

**Comment by Poorchop** [comment]
After chatting with some php "experts," I restored the default `session.save_path` settings and followed an old method that I used outlined [here](https://www.centos.org/forums/viewtopic.php?t=48750), namely setting the owner of the session folder to root:nginx. Now I'm not seeing any new messages in my Lychee error log, but I'm still receiving the permission error. I even tried setting the owner to nginx:nginx and I disabled SELinux just to test, but still no luck.

With root:nginx as the owner, I'm finally seeing the session file in /var/lib/php/session as expected, but that's as far as it gets and like I said, I still get the permission error but now I have no error messages with which I can work. I can't imagine what could still be causing permission errors especially with SELinux off - I set 777 on Lychee/data and Lychee/uploads. My /tmp folder still have the default permissions. Nginx is the owner of /var/www/Lychee and all subdirectories. What am I missing?

**Comment by Poorchop** [comment]
Took me an entire day to figure this out but I realized that my problem was with creating a mysql database. The session folder does indeed need to be owned by the httpd whether it be apache, nginx, or something else. I set some additional SELinux rules, explicitly allowing the httpd to connect to databases. Then I tried to see if I could created a database manually via the cli, and finally I tried doing it again via the web interface.

Uploading and viewing photos finally works, so I'm really happy about that. However, now I'm running into other issues with the plugins, so time to start tackling those problems.

For anyone running nginx and php-fpm on Fedora/CentOS, just keep in mind that it's packaged poorly because it assumes that the httpd is apache. Hopefully this helps someone in the future.
